### PR TITLE
Switch to use otel/trace instead of opencensus

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

**Comment by dmitryax** [comment]
needs rebase

**Comment by bogdandrutu** [comment]
@dmitryax done, PTAL

File: client/worker.go [hunk 1]
```diff
@@ -54,56 +61,46 @@ func newWorker(client *http.Client, endpoint string, accessToken string, disable
...
 	}
 	recordSendFailure(ctx, sr)
-	span.SetStatus(trace.Status{
-		Code: OCStatusCodeFromHTTP(int32(serr.StatusCode)),
-	})
 
+	span.SetStatus(codes.Error, serr.Err.Error())
 	if serr.Permanent {
 		recordDrops(ctx, sr)
+		span.SetStatus(codes.Error, "unable to prepare request")
```
Comment (dmitryax) [inline client/worker.go:96]: Is this correct? `serr` is error from `send` not `prepare`

Comment (bogdandrutu) [reply client/worker.go]: You are correct, and anyway we set the error 2 lines above. Removing.

File: client/worker.go [hunk 2]
```diff
@@ -54,56 +61,45 @@ func newWorker(client *http.Client, endpoint string, accessToken string, disable
...
-	sr, err := w.prepare(ctx, batches, spansCount)
+	sr, err := w.prepare(batches, spansCount)
 	if err != nil {
+		span.SetStatus(codes.Error, err.Error())
 		recordEncodingFailure(ctx, sr)
-		span.SetStatus(trace.Status{
-			Code: trace.StatusCodeInvalidArgument,
-		})
-		return &ErrSend{Err: err}
+		return &ErrSend{Err: err, Permanent: true}
```
Comment (dmitryax) [inline client/worker.go:83]: Adding `Permanent` seems like unrelated fix. Can you at least mention it to the PR/commit comment.

**Review by dmitryax — APPROVED** [review]
One nit. Otherwise LGTM

File: client/worker.go (hunk 2 above)
Comment (bogdandrutu) [reply client/worker.go]: https://github.com/signalfx/sapm-proto/pull/56 for this
